xt_training/utils/sklearn_wrapper.py

```diff
@@ -62,6 +69,82 @@ def __len__(self):
         return len(self.dataset)
 
 
+class SKDataLoader:
+    """DataLoader wrapper class to enable training scikit learn models with xt-training.
+
+    This data loader will set batch_size such that all data is returned in a single batch,
+    suitable for most SKLearn models when trained with xt-training. When using this class,
+    there is no need to wrap the dataset object with SKDataset first.
+
+    Usage:
+
+        >>> import time
+        >>> loader = SKDataLoader(
+        ...     dataset,
+        ...     shuffle=True,
+        ...     cache=True # true by default, but shown here for illustration
+        ... )
+        >>> start = time.time()
+        >>> x, y = next(iter(loader))
+        >>> print(time.time() - start)
+        10.609230041503906
+        
+        >>> # Subsequent calls use the cache
+        >>> start = time.time()
+        >>> x, y = next(iter(loader))
+        >>> print(time.time() - start)
+        0.011992454528808594
+    """
+
+    def __init__(
+        self, dataset, shuffle=False, num_workers=0, collate_fn=None, pin_memory=False, timeout=0,
+        worker_init_fn=None, multiprocessing_context=None, cache=True
+    ):
+        """Contructor for SKDataLoader class.
+
+        Args:
+            cache (bool, optional): Whether to store a cached version of the dataset to use for
```

Comment:
I have a question: if the variable can be more than one type, e.g. `int` or `list of int`, can we declare it in the argument declaration?

Reply:
Good question. Yes, you can using the `typing` library. It can be done like this:

```
from typing import Union, List

def func(arg: Union[int, List[int]]):
    ...
```